internal/lib/sandbox/sandbox.go

```diff
@@ -315,3 +321,27 @@ func (s *Sandbox) Ready(takeLock bool) bool {
 
 	return cState.Status == oci.ContainerStateRunning
 }
+
+// UnmountShm removes the shared memory mount for the sandbox and returns an
+// error if any failure occurs.
+func (s *Sandbox) UnmountShm() error {
+	if s.ShmPath() == DevShmPath {
+		return nil
+	}
+
+	// we got namespaces in the form of
+	// /var/run/containers/storage/overlay-containers/CID/userdata/shm
+	// but /var/run on most system is symlinked to /run so we first resolve
+	// the symlink and then try and see if it's mounted
+	fp, err := symlink.FollowSymlinkInScope(s.ShmPath(), "/")
+	if err != nil {
+		return err
+	}
+	if mounted, err := mount.Mounted(fp); err == nil && mounted {
+		if err := unix.Unmount(fp, unix.MNT_DETACH); err != nil {
+			return err
```

Comment:
Should we ignore EINVAL?
```
man umount
...
       EINVAL target is not a mount point.
```
This would happen if /dev/shm was already unmounted.

Reply:
we check if it's mounted before attempting to Unmount. Should protect against that case